src/targets/gpu/lowering.cpp

```diff
@@ -198,25 +198,17 @@ struct miopen_apply
             auto&& op                         = any_cast<Op>(ins->get_operator());
             auto beta                         = op.beta;
             std::vector<instruction_ref> refs = ins->inputs();
-            if((refs.size() == 2) or (refs.size() == 3 and refs.back()->outputs().size() > 1) or
-               (ins == last))
+            auto output                       = insert_allocation(ins, ins->get_shape());
+            if(refs.size() == 2)
             {
-                auto output = insert_allocation(ins, ins->get_shape());
-                if(refs.size() == 2)
-                {
-                    beta = 0;
-                    refs.push_back(output);
-                }
-                else
-                {
-                    auto copy_out = prog->insert_instruction(ins, hip_copy{}, refs.back(), output);
-                    refs.back()   = copy_out;
-                    refs.push_back(copy_out);
-                }
+                beta = 0;
+                refs.push_back(output);
             }
             else
             {
-                refs.push_back(refs.back());
+                auto copy_out = prog->insert_instruction(ins, hip_copy{}, refs.back(), output);
```

Comment:
This will always make a copy even if its only used once. Why dont you check for `refs.back()->outputs().size() > 1` anymore?